conductr_cli/sandbox_start.py

```diff
@@ -0,0 +1,159 @@
+    current_port_str_rev = ''.join(reversed(str(port)))
+    current_second_last_nr = int(current_port_str_rev[1])
+    new_second_last_nr = current_second_last_nr if instance == 0 else (current_second_last_nr + instance) % 10
+    new_port_str_rev = current_port_str_rev[0] + str(new_second_last_nr) + current_port_str_rev[2:]
+    return ''.join(reversed(new_port_str_rev))
+
+def inspect_cond0_ip():
+    return terminal.docker_inspect('{}0'.format(CONDUCTR_NAME_PREFIX), '{{.NetworkSettings.IPAddress}}')
+
+def resolve_conductr_roles_by_container(conductr_roles, instance):
+    if not conductr_roles:
+        # No ConductR roles have been specified => Return an empty Set
+        return set()
+    elif instance + 1 <= len(conductr_roles):
+        # Roles have been specified for the current ConductR instance => Get and use these roles.
+        return conductr_roles[instance]
+    else:
+        # The current ConductR instance is greater than the length of conductr_roles.
+        # In this case the roles of conductr_roles are subsequently applied to the remaining instances.
+        remainder = (instance + 1) % len(conductr_roles)
+        remaining_instance = len(conductr_roles) if remainder == 0 else remainder
+        return conductr_roles[remaining_instance - 1]
+
+def run_conductr_cmd(instance, container_name, cond0_ip, envs, image, log_level, ports, feature_names, conductr_roles):
+    general_args = ['-d', '--name', container_name]
+    env_args = resolve_docker_run_envs(envs, log_level, cond0_ip, feature_names, conductr_roles)
+    print('env_args: {}'.format(env_args))
+    port_args = resolve_docker_run_port_args(ports, instance)
+    optional_args = general_args + env_args + port_args
+    positional_args = resolve_docker_run_positional_args(cond0_ip)
+    terminal.docker_run(optional_args, image, positional_args)
+
+def resolve_docker_run_envs(envs, log_level, cond0_ip, feature_names, conductr_roles):
+    log_level_env = ['AKKA_LOGLEVEL={}'.format(log_level)]
+    syslog_ip_env = ['SYSLOG_IP={}'.format(cond0_ip)] if cond0_ip else []
+    conductr_features_env = ['CONDUCTR_FEATURES={}'.format(','.join(feature_names))] if feature_names else []
+    conductr_roles_env = ['CONDUCTR_ROLES={}'.format(','.join(conductr_roles))] if conductr_roles else []
+    all_envs = envs + log_level_env + syslog_ip_env + conductr_features_env + conductr_roles_env
+    env_args = []
+    for env in all_envs:
+        env_args.append('-e')
+        env_args.append(env)
+    return env_args
+
+def resolve_docker_run_port_args(ports, instance):
+    port_args = []
+    for port in ports | CONDUCTR_PORTS:
+        port_args.append('-p')
+        port_args.append(map_port(instance, port))
+    return port_args
+
+def resolve_docker_run_positional_args(cond0_ip):
+    seed_node_arg = ['--seed', '{}:{}'.format(cond0_ip, 9004)] if cond0_ip else []
+    return ['--discover-host-ip'] + seed_node_arg
+
+# Stop nodes
+def stop_nodes():
+    '''TODO: Implement'''
```

Comment:
Just wondering if you should seek any inspiration from the original run_conductrs python script we had - perhaps not, but just thought I'd remind you of it.